src/snowflake/snowpark/query_history.py

```diff
@@ -12,28 +12,90 @@ class QueryRecord(NamedTuple):
 
     query_id: str
     sql_text: str
+    is_describe: bool = None
+    thread_id: int = None
+
+    def __repr__(self) -> str:
+        if self.is_describe is None and self.thread_id is None:
+            return f"QueryRecord(query_id={self.query_id}, sql_text={self.sql_text})"
+        elif self.is_describe is not None and self.thread_id is None:
+            return f"QueryRecord(query_id={self.query_id}, sql_text={self.sql_text}, is_describe={self.is_describe})"
+        elif self.is_describe is None and self.thread_id is not None:
+            return f"QueryRecord(query_id={self.query_id}, sql_text={self.sql_text}, thread_id={self.thread_id})"
+        else:
+            return f"QueryRecord(query_id={self.query_id}, sql_text={self.sql_text}, is_describe={self.is_describe}, thread_id={self.thread_id})"
+
 
+class QueryListener:
+    @abstractmethod
+    def _notify(self, query_record: QueryRecord, *args, **kwargs) -> None:
```

Comment:
args is not really used now. Is it only done for future proofing? Do you foresee args to be used anytime soon?

Reply:
let's remove *args

Reply:
done.